src/main/java/pl/agawesolowska/ticketbookingapp/facade/BookingFacade.java

```diff
@@ -0,0 +1,81 @@
+package pl.agawesolowska.ticketbookingapp.facade;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import pl.agawesolowska.ticketbookingapp.model.dto.BookingRequestDTO;
+import pl.agawesolowska.ticketbookingapp.model.dto.BookingResultDTO;
+import pl.agawesolowska.ticketbookingapp.model.entity.Booking;
+import pl.agawesolowska.ticketbookingapp.model.entity.Customer;
+import pl.agawesolowska.ticketbookingapp.model.entity.Screening;
+import pl.agawesolowska.ticketbookingapp.model.entity.Seat;
+import pl.agawesolowska.ticketbookingapp.service.BookingService;
+import pl.agawesolowska.ticketbookingapp.service.CustomerService;
+import pl.agawesolowska.ticketbookingapp.service.ScreeningService;
+import pl.agawesolowska.ticketbookingapp.service.SeatService;
+import pl.agawesolowska.ticketbookingapp.util.DateTimeUtils;
+import pl.agawesolowska.ticketbookingapp.util.SeatReservationUtils;
+
+/**
+ * @author Aga Wesołowska
+ *
+ */
+@Component
+public class BookingFacade {
+
+	private final BookingService bookingService;
+	private final CustomerService customerService;
+	private final ScreeningService screeningService;
+	private final SeatService seatService;
+
+	@Autowired
+	public BookingFacade(BookingService bookingService, CustomerService customerService,
```

Comment:
kiedy klasa ma za dużo zależności?

Reply:
Myślę, że powyżej 3. Ta fasada jest 'przesadzona'. Dopóki nie sprawdzałam czy rezerwacja jest możliwa ze względu na datę/godzinę rozpoczęcia seansu były trzy zależności: Booking/Customer/SeatService. Trochę podobnie jak z metodami 3-argumentowymi, tak jak pierwsza wersja kontrolera (commit z początku kwietnia 881e24b) - nie podobało mi się to rozwiązanie, było nieczytelne, dlatego zaczęłam kombinować z fasadą i DTO.